lib/core.rb

```diff
@@ -11,11 +11,7 @@ def initialize
 
   def guess(number)
     @statistics[:tries] += 1
-    case number.to_i <=> @secret
-    when -1 then :higher
-    when 0 then :win # end of game flag
-    when 1 then :lower
-    end
+    { -1 => :higher, 0 => :win, 1 => :lower }[number.to_i <=> @secret]
```

Comment:
This is readable.  I am thinking that aside from the key ":win" symbol, in another language `:lower` and `:higher` may be different.  So this depends on if you are going to explore translations.  As it is, we let the perfect wait, when the code is working and is indeed good enough.